lib/TemplatedPathPlugin.js

```diff
@@ -53,10 +53,19 @@ const getReplacer = (value, allowEmpty) => {
 	return fn;
 };
 
+const escapePathVariables = value => {
+	return typeof value === "string"
+		? value.replace(
+				/\[(name|id|moduleid|file|query|filebase|url|hash|chunkhash|modulehash|contenthash)\]/g,
```

Comment:
```suggestion
				/\[(\\*[\w:]+\\*)\]/gi,
```

To be future proof we allow word characters and the colon in `[]`. So this need to be escape.

There is also a double escape trap, i. e. you want to have `[\abc\]` as plaintext. So this must also be escaped, otherwise it would be unescaped and incorrectly lead to `[abc]`. Escaping and unescaping must be perfectly reversible.